x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/types.ts

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export interface IInferencePipeline {
+  pipelineName: string;
+  trainedModelName: string;
+  isDeployed: boolean;
+  modelTypes: string[];
+}
```

Comment:
I think for consistency's sake, we want to remove the first `I` from the interface name. Then `modelTypes` seems to be a combination of `model_type` and `tags` fields from the [Get trained models API](https://github.com/elastic/elasticsearch/edit/8.4/docs/reference/ml/trained-models/apis/get-trained-models.asciidoc). Not sure if you want both or a specific one.
```TypeScript
export interface InferencePipeline {
  isDeployed: boolean;
  modelType: string;
  pipelineName: string;
  tags: string[];
  trainedModelName: string;
}
```